llvm/lib/Transforms/Scalar/InductiveRangeCheckElimination.cpp

```diff
@@ -279,6 +279,9 @@ bool InductiveRangeCheck::parseRangeCheckICmp(Loop *L, ICmpInst *ICI,
   Value *LHS = ICI->getOperand(0);
   Value *RHS = ICI->getOperand(1);
 
+  if (!LHS->getType()->isIntegerTy())
```

Comment:
Better to use `SE->isSCEVable(LHS->getType())` as it makes the intention more explicit?

Reply:
Pointer type is SCEVable, but `willNotOverflow` only supports binops with integers.

Reply:
Ah my bad, thanks for clarifying